### DISCUSSION TITLE
Can I Render a template from a database?

### DISCUSSION THREAD

**Discussion by michaelakin** [soundaranbu/RazorTemplating#79] [Q&A] [answered]
Hi, I was wondering a couple of things. 

1) how can I utilize this engine to render from the template that is in my database and not a fixed view?
1a)  Even if it was a fixed view, could I do two passes of rendering?  1) to insert my "template" into the razor view, and 2) to insert data into the template I inserted.

2)  What is the difference of this and the new HtmlRenderer  ?

**Comment by soundaranbu** [comment, marked as answer]
Hello, 
1. Yes, it's possible. You might want to take a look at this sample project https://github.com/soundaranbu/Razor.Templating.Core/tree/feature/custom-razor-file-provider/examples/_RealWorldSamples/RazorTemplateEditor where it's saving the view in the database and then uses it to render the template based on the given input
2. The new [HtmlRenderer introduced in .NET 8](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/render-components-outside-of-aspnetcore?view=aspnetcore-9.0) is based on razor components(.razor) whereas this library utilizes the View files(.cshtml) with concepts similar to MVC but allows it render anywhere outside of the HttpContext (except the browser)

I hope that answers your question!

**Reply by michaelakin** [reply]
Thanks for the reply.  I will check out the link you listed.
